server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java

```diff
@@ -91,24 +88,25 @@ public void updateCacheForAllPipelinesIn(CruiseConfig config) {
     }
 
     private GoDashboardPipelineGroup dashboardPipelineGroupFor(PipelineConfigs pipelineGroup, PipelineSelections pipelineSelections, Username user) {
-        GoDashboardPipelineGroup goDashboardPipelineGroup = new GoDashboardPipelineGroup(pipelineGroup.getGroup(), resolvePermissionsForPipelineGroup(pipelineGroup));
+        GoDashboardPipelines allPipelines = cache.allEntries();
```

Comment:
You should really be doing this once, in `allPipelineGroupsForDashboard` method.